builtin/init-db.c

```diff
@@ -258,15 +258,34 @@ static int create_default_files(const char *template_path,
 		die("failed to set up refs db: %s", err.buf);
 
 	/*
-	 * Create the default symlink from ".git/HEAD" to the "master"
-	 * branch, if it does not exist yet.
+	 * Create the default symlink from ".git/HEAD" to the default
+	 * branch name, if it does not exist yet.
 	 */
 	path = git_path_buf(&buf, "HEAD");
 	reinit = (!access(path, R_OK)
 		  || readlink(path, junk, sizeof(junk)-1) != -1);
 	if (!reinit) {
-		if (create_symref("HEAD", "refs/heads/master", NULL) < 0)
+		const char *branch_name;
+		char *prefixed;
+
+		/* get the default branch name from config, or failing that, env */
+		if (git_config_get_string_const("init.defaultbranchname", &branch_name))
+			branch_name = getenv("GIT_DEFAULT_BRANCH_NAME");
```

Comment:
We need to use the `GIT_TEST_` prefix here, otherwise the value will simply be unset by our test suite (meaning: we could not adjust the test suite incrementally).

Please use `GIT_TEST_DEFAULT_BRANCH_NAME` instead.